raiden/network/proxies/token_network.py

```diff
@@ -1071,81 +1067,66 @@ def _update_preconditions(
-            'node': pex(self.node_address),
-            'partner': pex(partner),
-            'nonce': nonce,
-            'balance_hash': encode_hex(balance_hash),
-            'additional_hash': encode_hex(additional_hash),
-            'closing_signature': encode_hex(closing_signature),
-            'non_closing_signature': encode_hex(non_closing_signature),
-        }
-        log.debug('updateNonClosingBalanceProof called', **log_details)
+            if partner_recovered_address != partner:
+                raise RaidenUnrecoverableError('Invalid update transfer signature')
 
-        checking_block = self.client.get_checking_block()
+        # Check the preconditions for calling updateNonClosingBalanceProof at
+        # the time the event was emitted.
         try:
-            self._update_preconditions(
+            channel_onchain_detail = self._detail_channel(
+                participant1=self.node_address,
+                participant2=partner,
+                block_identifier=given_block_identifier,
                 channel_identifier=channel_identifier,
-                partner=partner,
-                balance_hash=balance_hash,
-                nonce=nonce,
-                additional_hash=additional_hash,
-                closing_signature=closing_signature,
+            )
+            closer_details = self._detail_participant(
+                channel_identifier=channel_identifier,
+                participant=partner,
+                partner=self.node_address,
                 block_identifier=given_block_identifier,
             )
-        except NoStateForBlockIdentifier:
-            # If preconditions end up being on pruned state skip them. Estimate
-            # gas will stop us from sending a transaction that will fail
+        except ValueError:
+            # If `given_block_identifier` has been pruned the checks cannot be
+            # performed.
             pass
+        else:
+            if channel_onchain_detail.channel_identifier != channel_identifier:
+                msg = (
+                    f'The provided channel identifier does not match the value '
+                    f'on-chain at the provided block ({given_block_identifier}). '
+                    f'This call should never have been attempted. '
+                    f'provided_channel_identifier={channel_identifier}, '
+                    f'onchain_channel_identifier={channel_onchain_detail.channel_identifier}'
+                )
+                raise RaidenUnrecoverableError(msg)
+
+            if channel_onchain_detail.state != ChannelState.CLOSED:
+                msg = (
+                    f'The channel was not closed at the provided block '
+                    f'({given_block_identifier}). This call should never have '
+                    f'been attempted.'
+                )
+                raise RaidenUnrecoverableError(msg)
+
+            if channel_onchain_detail.settle_block_number < self.client.block_number():
```

Comment:
I think this rpc call is unnecessary, but I haven't got to a point where I can remove it yet.

Reply:
Humm, yeah, this is the wrong precondition check. Per the definition:

        Used to check if there are any logic bugs in the business logic
        which resulted in an impossible ContractSend*.

We should not test this against the current block, but against the block number at which we sent the transaction.